datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/counter.py

```diff
@@ -0,0 +1,516 @@
+            self.name,
+        )
+
+
+class CounterBase:
+    def __init__(
+        self, name, config, check, connection, object_name, metric_prefix, counter_selector, tags, custom_transformer
+    ):
+        self.logger = check.log
+        self.connection = connection
+        self.counter_selector = counter_selector
+        self.tags = tags
+
+        # The name of the counter e.g. System Calls/sec
+        self.name = name
+
+        # The name of the performance object e.g. System
+        self.object_name = object_name
+
+        self.metric_name = config.get('metric_name', '')
+        if not isinstance(self.metric_name, str):
+            raise ConfigTypeError(
+                f'Option `metric_name` for counter `{self.name}` of performance object '
+                f'`{self.object_name}` must be a string'
+            )
+        elif not self.metric_name:
+            metric_suffix = config.get('name', '')
+            if not isinstance(metric_suffix, str):
+                raise ConfigTypeError(
+                    f'Option `name` for counter `{self.name}` of performance object '
+                    f'`{self.object_name}` must be a string'
+                )
+            elif not metric_suffix:
+                raise ConfigValueError(
+                    f'Option `name` for counter `{self.name}` of performance object `{self.object_name}` is required'
+                )
+
+            self.metric_name = f'{metric_prefix}.{metric_suffix}'
+
+        if custom_transformer is not None:
+            self.metric_type = 'custom'
+            self.transformer = custom_transformer(check, self.metric_name, config)
+        else:
+            self.metric_type = config.get('type', 'gauge')
+            if not isinstance(self.metric_type, str):
+                raise ConfigTypeError(
+                    f'Option `type` for counter `{self.name}` of performance object '
+                    f'`{self.object_name}` must be a string'
+                )
+            elif self.metric_type not in TRANSFORMERS:
+                raise ConfigValueError(
+                    f'Unknown `type` for counter `{self.name}` of performance object `{self.object_name}`'
+                )
+
+            try:
+                self.transformer = TRANSFORMERS[self.metric_type](check, self.metric_name, config)
+            except Exception as e:
+                error = (
+                    f'Error compiling transformer for counter `{self.name}` of performance object '
+                    f'`{self.object_name}`: {e}'
+                )
+                raise type(e)(error) from None
+
+    def handle_counter_value_error(self, error, instance=None):
+        # Counter requires at least 2 data points to return a meaningful value, see:
+        # https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhgetformattedcountervalue#remarks
+        #
+        # http://timgolden.me.uk/pywin32-docs/error.html
+        if error.strerror != 'The data is not valid.':
```

Comment:
https://github.com/mhammond/pywin32/blob/4e488b08a54c7c8e8c4c8eb1506c9f28b5bbe6ff/win32/src/win32pdhmodule.cpp#L645 gives me:

```
pywintypes.error: (-1073738810, 'GetFormattedCounterValue', 'The data is not valid.')
```

Reply:
From the [doc](https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhgetformattedcountervalue#return-value) it seems like it can return other errors, should this handle those too?

Reply:
No b/c I think this is the only one that indicates it needs a 2nd value

Reply:
Ok sounds good, one more small question- is the log line `Waiting on another data point for instance `%s` of counter `%s` of performance object `%s` appropriate in those cases though?

Reply:
should be, why?

Reply:
Ok sounds good, I was just making sure I was understanding it correctly/